client/src/main/scala/io/mediachain/client/MediachainCopycatClient.scala

```diff
@@ -0,0 +1,155 @@
+package io.mediachain.client
+
+import cats.data.XorT
+import io.mediachain.protocol.Datastore.{JournalEntry, _}
+import io.mediachain.protocol.Transactor.JournalListener
+import io.mediachain.copycat.Client.{ClientState, ClientStateListener}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class MediachainCopycatClient(datastore: Datastore)
+  (implicit executionContext: ExecutionContext = ExecutionContext.global)
+  extends MediachainClient with JournalListener with ClientStateListener
+{
+
+  def allCanonicalReferences = canonicalRefs
+
+  def chainForCanonical(ref: Reference): Future[Option[Reference]] = {
+    // TODO: handle disconnected cluster state
+    cluster.lookup(ref)
+  }
+
+  def addListener(listener: ClientEventListener): Unit = {
+    listeners += listener
+  }
+
+
+  /**
+    * Add a new CanonicalRecord to the system
+    *
+    * @param canonicalRecord a new Entity or Artefact to add to the system
+    * @return an Xor-wrapped future that will complete with either a ClientError or a
+    *         Reference to the new record
+    */
+  def addCanonical(canonicalRecord: CanonicalRecord) =
+    XorT[Future, ClientError, Reference] {
+      cluster.insert(canonicalRecord)
+        .map(_.bimap(ClientError.Journal, _.ref))
```

Comment:
Another point here -- do we really want to wrap this in a different kind of error?
Why not use the JournalError directly? It is defined in the Transactor types after all.

I see yusef also wants to have a NetworkError, but perhaps we should extend the error hierarcy in Transactor instead of wrapping.